src/extension.ts

```diff
@@ -96,18 +166,37 @@ class PyPICodeLens extends vscode.CodeLens {
 
 class PyPICodeLensProvider implements vscode.CodeLensProvider<PyPICodeLens> {
     provideCodeLenses(document: vscode.TextDocument): PyPICodeLens[] {
+        const codeLensEnabled = vscode.workspace.getConfiguration('pypiAssistant').get('codeLens')
         const codeLenses: PyPICodeLens[] = []
-        if (vscode.workspace.getConfiguration('pypiAssistant').get('codeLens')) {
+
+        if (!codeLensEnabled) {
+            return codeLenses
+        }
+
+        if (document.languageId === 'toml' && document.fileName.toLowerCase().endsWith('pyproject.toml')) {
+            const dependencies = getPyProjectDependencies(document.getText())
+
             for (let line = 0; line < document.lineCount; line++) {
-                let requirement: Requirement | null
-                try {
-                    requirement = parsePipRequirementsLineLoosely(document.lineAt(line).text)
-                } catch {
-                    continue
-                }
+                const lineText = document.lineAt(line).text
+                const requirement = dependencies.find((dependency) => {
+                    return lineText.startsWith(`${dependency.name} = `)
```

Comment:
This is a bit wonky, because AFAIK TOML allows no spaces around the equals sign, as well as multiple spaces, which clashes with this template string. Another point for the `lineText.split('=')[0].trim()` workaround (…or the more complex AST parsing idea)